test_runner.py

```diff
@@ -0,0 +1,104 @@
+            raise ComplianceException(f'{message} Schema validation failed for {self.job_data["operation"]} '
+                                      f'{self.job_data["endpoint"]}. Error Details - {err}')
+
+    def validate_request_body(self, request_body):
+
+        # JSON Validation
+        try:
+            request_body_json = json.loads(request_body)
+        except json.JSONDecodeError as err:
+            raise ComplianceException(f'JSON Error in request body for '
+                                      f'{self.job_data["operation"]} {self.job_data["endpoint"]} - {err}')
+
+        # Logical Schema Validation
+
+        endpoint_model = self.job_data["name"] + "_request_body"
+        self.validate_logic(endpoint_model, request_body_json, "Request Body")
+
+    def validate_response(self, response):
+
+        # General status validation
+        response_status = list(self.job_data["response"].keys())[0]
+
+        if response.status_code == response_status:
+            print(f'{self.job_data["operation"]} {self.job_data["endpoint"]} Successful Response status code')
+        else:
+            raise ComplianceException(f'{self.job_data["operation"]} {self.job_data["endpoint"]}'
+                                      f' Response status code is not 200')
+
+        # Logical Schema Validation
+        if not response.text:
+            response_json = {}          # Handle the Cancel Task Endpoint empty response
+        else:
+            response_json = response.json()
+        # print(response_json)
+
+        if self.job_data["name"] in ["list_tasks", "get_task"]:
+            view_query = [item["view"] for item in self.job_data["query_parameters"]]
+            endpoint_model = self.job_data["name"] + "_" + view_query[0]
+        else:
+            endpoint_model = self.job_data["name"]
+        self.validate_logic(endpoint_model, response_json, "Response")
+        self.auxiliary_space[self.job_data["name"]] = response_json
+
+    def run_tests(self, job_data):
+
+        # print(job_data)
+        self.job_data = job_data
+
+        id_uri_param = None
+        query_params = {}
+        request_body = None
+
+        if job_data["name"] in ["get_task", "cancel_task"]:
+            id_uri_param = self.auxiliary_space["create_task"]["id"]
+
+        if job_data["name"] in ["get_task", "list_tasks"]:
+            for param in job_data["query_parameters"]:
+                query_params.update(param)
+            # print(query_params)
+
+        if job_data["name"] in ["create_task"]:
+            request_body = job_data["request_body"]
+            self.validate_request_body(request_body)
+            # print(request_body)
+
+        if "polling" in job_data.keys():
+
+            check_cancel = False
+            # print(self.auxiliary_space)
+            if "cancel_task" in self.auxiliary_space.keys():
+                check_cancel = True
+
+            response = poll_request(self.server, self.version, job_data["endpoint"],
```

Comment:
To make your code easier to read, please add the argument names in function calls (at least if there are more than one and it's not a very well known built-in function). It's very difficult to track (and also error-prone) if you pass multiple arguments by position, especially if the functions/methods are not typed